yang-module-versioning/draft-ietf-netmod-yang-module-versioning.xml

```diff
@@ -434,6 +446,129 @@ revision 2019-01-02 {
+    <t>The following example illustrates the schema node compatibility extensions:
+      <figure>
+        <artwork>
+          container some-stuff {
+            leaf used-to-be-a-string {
+              rev:node-non-backwards-compatible-change "3.0.0";
+              type uint32;
+            }
+            leaf fixed-my-description-typo {
+              rev:node-editorial-change "1.4.1";
+              type string;
+              description "This description used to have a typo."
+            }
+            list sir-changed-a-lot {
+              rev:node-editorial-change "3.0.0";
+              rev:node-backwards-compatible-change "2.3.0";
+              rev:node-backwards-compatible-change "1.2.1_non_compatible";
+              description "a list of stuff";
+              ordered-by user;
+              key "foo";
+              leaf foo {
+                type string;
+              }
+              leaf thing {
+                type uint8;
+              }
+            }
+        </artwork>
+      </figure>
+    </t>
+    <t>Note that an individual schema node may have a backwards compatible change
+    in a revision that is non-backwards-compatible (e.g. some other node changed
+    in a non-backwards-compatible fashion in that particular revision).
+    </t>
+    <t>If changes are ported from one branch of YANG model revisions to another
+    branch, care must be taken with any schema node compatibilty extensions. 
+    A simple copy-n-paste should not be used. The extensions 
+    may incorrectly reference a revision that is not in the history of the 
+    new revision. Further, the annotations might not apply depending on what 
+    the history is like in that new branch (e.g., an NBC change
+    that is ported might not be an NBC change in the new branch).
+    Schema node compatiblity extensions should not be copied over to the new 
+    branch. Instead, the changes should be 
+    considered as completely new on the new branch, and any compatibility information
+    should be generated from scratch.
+    </t>
+    <t>TBD: Is the format right in the example above ? quotes and semi colon ?</t>
+    <t>TBD: Are we going to define when to add and not add these tags ? e.g.
+    optional anytime they are applicable, mandatory when a tool can't guess ?</t>
+    <t>
+    TBD: can the backwards-compatible-node tag be used in blatant contradiction to NBC rules ?  E.g. a range shrinks from 1..50 to 0..9. Is an author allowed to tag that leaf with the backwards-compatible-node tag ? Then what happens with the YANG Semver and the 
+    top level revision NBC tag ?
+    </t>
+    <t>
+    TBD: use the same non-backwards-compatible tag as the top level, but allow it to be applied to nodes ?
+    </t>
+    <t>
+    TBD: The scheme comparison draft limited the use of per-node tags to specific
+    YANG statements: description, must and when (and probably 'pattern' should
+    have been in that list). Are we going to limit their applicability ?
```

Comment:
It may be that we broaden this to address WG comments, but recommend judicious use to those situations where the NBC/BC change is not so terribly obvious.

Reply:
The per-node tags are really per-YANG-statement tags but for a subset of YANG statements (probably "data nodes" + rpc, notification, action, etc).
But we do probably want these tags to apply to choice, case, and grouping.